website/registrations/utils.py

```diff
@@ -0,0 +1,376 @@
+                    try:
+                        full_name = match.group('full_name')
+                        email = match.group('email')
+                    except AttributeError:
+                        self.log_error(invalid=True)
+                    else:
+                        parsed_value.append({'full_name': full_name, 'email': email})
+            self._parsed_value = parsed_value
+
+class LicenseField(MetadataField):
+    # format: license_name;year;copyright_holder_one,copyright_holder_two,...
+    with_required_fields_regex = re.compile(r'(?P<name>[\w ]+);(?P<year>[ ][1-3][0-9]{3});(?P<copyright_holders>[\w -,]+)')
+    no_required_fields_regex = re.compile(r'(?P<name>[\w ]+)')
+
+    def _validate(self):
+        parsed_value = None
+        if self.required and not bool(self.value):
+            self.log_error(missing=True)
+        else:
+            license_name_match = self.no_required_fields_regex.match(self.value)
+            if license_name_match is not None:
+                node_license_name = license_name_match.group('name')
+                try:
+                    node_license = NodeLicense.objects.get(name__iexact=node_license_name)
+                except NodeLicense.DoesNotExist:
+                    self.log_error(invalid=True)
+                else:
+                    has_required_fields = bool(node_license.properties)
+                    if has_required_fields:
+                        match = self.with_required_fields_regex.match(self.value)
+                        if match is not None:
+                            name = match.group('name')
+                            year = match.group('year')
+                            copyright_holders = match.group('copyright_holders')
+                            copyright_holders = [val.strip() for val in copyright_holders.split(',')]
+                            parsed_value = {'name': name,
+                                            'required_field': {'year': year,
+                                                               'copyright_holders': copyright_holders}}
+                    else:
+                        parsed_value = {'name': node_license_name}
+                    self._parsed_value = parsed_value
+            else:
+                self.log_error(invalid=True)
+
+class CategoryField(MetadataField):
+    def _validate(self):
+        try:
+            self._parsed_value = settings.NODE_CATEGORY_MAP[self.value]
+        except KeyError:
+            self.log_error(invalid=True)
+
+class SubjectsField(MetadataField):
+    def _validate(self):
+        subjects = [val.strip() for val in self.value.split(';')]
+        valid_subjects = list(Subject.objects.filter(text__in=subjects).values_list('text', flat=True))
+        invalid_subjects = list(set(subjects) - set(valid_subjects))
+        if len(invalid_subjects):
+            self.log_error(invalid=True)
+        else:
+            self._parsed_value = valid_subjects
```

Comment:
A note for the creation task: only the leaf-level subjects are provided and the task needs to apply all of its parent subjects.